pkg/apis/kops/secret.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kops
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient=true
+
+// Secret represent a set of kops secrets
+type Secret struct {
```

Comment:
I don't think we should call it Secret just for confusion (I think they are namespaced by the group, so it is technically permitted, but all the same it is confusing).  Perhaps `SystemSecret` or `SecretMaterial` or `ClusterSecret` or something like that...  I did `KeySet` before which I think is fine but weaker than the 3 I just mentioned.  WDYT?